components/Register/RegisterPage.js

```diff
@@ -0,0 +1,118 @@
+            const data = {
+                email: userEmail,
+                firstName: userFirstName,
+                lastName: userLastName,
+                password: userPassword,
+                phoneNumber: userPhonerNumber,
+            };
+            axios.post('/api/v1/user', data)
+                .then(() => {
+                    setMessage('Registration Successful. You can Login');
+                    setEmail('');
+                    setFirstName('');
+                    setLastName('');
+                    setPassword('');
+                    setphoneNumber('');
+                })
+                .catch(error => {
+                    setMessage(error);
+                });
+        }
+    };
+
+    return (
+        <div>
+            <Grid container component="main" className="root">
+                <CssBaseline />
+                <Grid item xs={false} sm={4} md={7} className="image" />
+                <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
+                    <div className="paper">
+                        <Typography component="h1" variant="h5">
+                            Sign in
+                        </Typography>
+                        <form className="form" noValidate>
+                            <Grid container>
+                                <Grid item>
+                                    <h2>
+                                        {'Become a Member'}
+                                    </h2>
+                                </Grid>
+                            </Grid>
+                            <Grid container>
+                                <Grid item>
+                                    <h4>
+                                        {message
+                                            ? 'Your Registration was Successfuls. You can Login '
+                                            : ''}
+                                    </h4>
+                                </Grid>
+                            </Grid>
+                            {
+                                TEXT_FIELD.map(({
+                                    label, name, id, type, autoFocus, value, handler,
+                                }) => (
+                                    <TextField
+                                      key={id}
+                                      variant="outlined"
+                                      margin="normal"
+                                      required
+                                      fullWidth
+                                      label={label}
+                                      name={name}
+                                      type={type}
+                                      autoFocus={autoFocus}
+                                      value={value}
+                                      onChange={handler}
+                                      /* i used {e => {handler}(e.target.value)} but show error */
```

Comment:
Yes this `{e => {handler}(e.target.value)} ` will throw an error because handler here is a string not a method.